common/src/main/java/com/wynntils/features/user/ChangelogFeature.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.net.ApiResponse;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.screens.changelog.ChangelogScreen;
+import com.wynntils.utils.mc.McUtils;
+import java.util.Map;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChangelogFeature extends UserFeature {
+    @Config(visible = false)
+    public String lastShownVersion = "v0.0.2-alpha.1";
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        if (event.isFirstJoinWorld()) {
+            ApiResponse response = Managers.Net.callApi(
+                    UrlId.API_ATHENA_UPDATE_CHANGELOG,
+                    Map.of("old_version", lastShownVersion, "new_version", WynntilsMod.getVersion()));
+
+            response.handleJsonObject(
+                    jsonObject -> {
+                        if (!jsonObject.has("changelog")) return;
+
+                        String changelog = jsonObject.get("changelog").getAsString();
+
+                        lastShownVersion = WynntilsMod.getVersion();
+
+                        Managers.TickScheduler.scheduleNextTick(
```

Comment:
I think this is a bit scary, throwing the changelog in your face when you log in, and might be in a vulnerable situation.

 Legacy had some thing where you did not immediately did not connect if you pressed the W button on the main screen, and there were an update pending, instead it showed the changelog instead of logging in.

I have two suggestions:
1) Let the user login, but if we have updates and want to show a changelog, throw them over to `/hub`, unless they already are, first.

2) See if we can show the changelog at the main screen instead

3) Intercept the wynncraft server connection with a mixin and abort it if we want to show a changelog.

Reply:
We do not do any updating? This feature is to show the updates between last played and current version. I get what you are saying, but /hub-ing the user is undesirable behavior too.

Reply:
No, but we throw a screen in the face of the user, at a point in time where they might not be able to handle it.

Reply:
@magicus Try it now :). I think it works neatly :).

Reply:
Ah, `/class` is fine, and better than hub! Good thinking. I assume you have tested both with and without our special class selection screen?

Reply:
Havent tested without our screen, but that was the one giving problems anyways. I accounted for both options though